packages/medusa/src/api/admin/users/[id]/route.ts

```diff
@@ -71,10 +74,10 @@ export const DELETE = async (
   res: MedusaResponse<HttpTypes.AdminUserDeleteResponse>
 ) => {
   const { id } = req.params
```

Comment:
We need to check here if the `id` matches the token `actor_id`, otherwise anyone can delete any customer account. When we have roles this might be ok (eg. user A ca delete user B if user A is an admin), but I am not sure if we want to allow that for now, WDYT?

Reply:
I think both options are not great actually.

Without the ability to delete users, you can't remove an unwanted admin user. 
With the ability to delete users, the unwanted admin user can delete you.

Reply:
Updated + added a test 👍

Reply:
Yeah indeed, I guess for now the one that has access to the DB has the ultimate authority (so they can remove someone manually), but the same goes for restoring access when someone removes you, so there is no right or wrong here, I leave it up to you to decide which one you think is better.